src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,51 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+
+        if (user == null) {
+            throw new InvalidDataException("User cannot be null");
+        }
+        if (Storage.people.contains(storageDao.get(user.getLogin()))) {
+            throw new InvalidDataException("User already exists "
+                    + user.getLogin()
+                    + " with such login");
+        }
+        if (user.getLogin() == null) {
+            throw new InvalidDataException("User login cannot be null");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new InvalidDataException("Not valid login." + "Min length is " + MIN_LENGTH);
+        }
+        if (user.getPassword() == null) {
+            throw new InvalidDataException("User password cannot be null");
+        }
+        if (user.getPassword().length() < 6) {
+            throw new InvalidDataException("Not valid password." + "Min length is " + MIN_LENGTH);
+        }
```

Comment:
The password length validation message should be more user-friendly. Consider rephrasing it to something like: 'Password must be at least ' + MIN_LENGTH + ' characters long.'